projects/bp-gallery/src/views/curator/UnlinkCollectionAction.tsx

```diff
@@ -0,0 +1,71 @@
+import { Icon, Tooltip } from '@mui/material';
+import React, { useCallback, useContext } from 'react';
+import { FlatCollection } from '../../types/additionalFlatTypes';
+import { DialogContext, DialogPreset } from '../shared/DialogWrapper';
+import { useUpdateCollectionMutation } from '../../graphql/APIConnector';
+import { useTranslation } from 'react-i18next';
+
+const UnlinkCollectionAction = ({
+  childCollection,
+  parentCollection,
+}: {
+  childCollection: FlatCollection;
+  parentCollection: FlatCollection;
+}) => {
+  const { t } = useTranslation();
+  const dialog = useContext(DialogContext);
+  const [updateCollection] = useUpdateCollectionMutation({
+    refetchQueries: ['getCollectionInfoById', 'getAllCollections'],
+    errorPolicy: 'all',
+  });
+
+  const onUnlinkChildCollection = useCallback(
+    (collection: FlatCollection) => {
+      dialog({
+        preset: DialogPreset.CONFIRM,
+        title: t('curator.reallyUnlink'),
+        content: t('curator.unlinkFromCollection', {
+          parent: parentCollection.name,
+        }),
+      }).then(resp => {
+        if (!resp) {
+          return;
+        }
+        const newParents = collection.parent_collections?.map(p => p.id) ?? [];
+        newParents.splice(newParents.indexOf(parentCollection.id), 1);
```

Comment:
I think filtering the `parentCollection.id` out of the `newParents` array would be more clean here, what do you think?